src/ci/docker/x86_64-gnu-tools/checktools.sh

```diff
@@ -88,7 +89,6 @@ status_check() {
     # these tools are not required for beta to successfully branch
     check_dispatch $1 nightly miri src/tools/miri
     check_dispatch $1 nightly embedded-book src/doc/embedded-book
-    check_dispatch $1 nightly rustc-guide src/doc/rustc-guide
```

Comment:
Put this back. This line only verifies that if you have updated `rustc-guide` the test must pass, and the `nightly` argument already excluded it from being checked in the beta week.

Reply:
I feel it would be beneficial to remove it.  If I include `rustc-guide` in an update, but it fails due to some transient error, I would prefer if that didn't fail the build.  Generally, I think bad links in `rustc-guide` should not prevent it from being updated.

Reply:
Disagreed. 

If it is really a transient failure you could just `@bors retry`. 

If the link is an actual bad link it could simply be updated or removed.

Reply:
It takes time to investigate why something failed. It involves loading log files. It involves investigating if it is transient or legitimate. It requires time to remove from a PR, and reporting issues.  These are all things I do not want to do.  Particularly for rustc-guide, which is not really published from here (I love rustc-guide BTW, don't get me wrong).  No other documentation checks external links, and I don't think it is a good idea to gate on it.

If it is enforced, I probably won't include rustc-guide in book updates. Someone else can submit updates for it.  I apologize for sounding curt, but I just don't want to spend my time on it.

Reply:
If we don't care about whether a link failed why do we add a linkchecker? Ignoring the alarm isn't good.

Besides, not gating on it is even worse for debugging since the maintainer of rustc-guide would now need to dig the log archive to get the error message. While with gating at least the Rust-Log-Analyzer could help you extract the relevant logs. If you don't want to investigate you could just mention the rustc-guide maintainer. Since other books are already gated on doc-tests you do need to investigate the log to find out which book is broken anyway.

Furthermore you seem to be exaggerating the probability of spurious external link failure.

Reply:
I don't know why the link checker was added. @mark-i-m said they already check it via cron, so I don't see what value it adds.  The very first PR that checked it failed. That is not a good indicator of how it will go.  It also looks like @mark-i-m has been actively combating CI problems with the link checker on the rustc-guide repo, which is also not encouraging.  I also cannot test it locally before posting if it only runs on Linux.  I've also had past experience dealing with checks like this, and it was quite common to have issues like [385](https://github.com/rust-lang/rustc-guide/issues/385).

If rustc-guide is already tested in cron, then @mark-i-m already knows when they get broken, probably much earlier than a submodule update will.  And if rustc-guide is tested on every PR on rust-lang/rust, then it will get 4-5 chances every day via the regression check to get alerted.  I don't think blocking a submodule update will provide any new signal.  Its only effect will be to take up my time.

Also, I'm usually not finding [the log message in RLA comments](https://github.com/rust-lang/rust/pull/62733#issuecomment-512605741) helpful.  It has been extremely rare where it showed the correct error for me.  

If you feel strongly about it, I'm willing to give it a trial run.  But if it takes up my time, I will probably remove it from my update list.

Reply:
AFAIK without this line, there's basically no toolstate tracking happening at all, or is there?

Reply:
@ehuss We added it because it is extremely helpful to know what PR caused a link to break (i.e. what API changed and how).

I'm personally a bit torn about gating update PRs. Thinking back on it now, not gating update PRs on this might be less helpful because the status would just remain broken if there is actually a problem. So someone would need to make another PR updating it again when it is fixed. On the other hand, our CI fail rate sue to transient failures is not great ATM, though I think it is better than @ehuss's expectations.

On the whole, I'm willing to just leave this PR as is, and I can ipgrade the check later once we see what happens.

Reply:
> AFAIK without this line, there's basically no toolstate tracking happening at all, or is there?

The status is saved in the local toolstate file by the call to `x.py test` at the top of the file (`save-toolstates` is set in `config.toml`).  The publishing is done at the bottom (`commit_toolstate_change`).

Reply:
So @ehuss @mark-i-m how can we proceed here? Personally I agree with @kennytm and, as @mark-i-m said, "not gating update PRs on this might be less helpful because the status would just remain broken". If a PR updates this submodule and tests still don't pass, surely something is wrong with that PR?

If you have spuriously failing tests, the [precedent with RLS](https://github.com/rust-lang/rust/issues/62225) is to disable/ignore those tests on your side so that at least the other tests can still do their job.

Reply:
As I said, I'm fine adding it back and seeing how it goes. I just don't want to spend much time dealing with failures going forward.  

I personally don't think a failure indicates a problem with the PR.  It just indicates the problem hasn't been fixed, yet.  A PR that updates a broken rustc-guide when it remains broken doesn't make the problem *worse*, it's already broken.  Making me take time to remove it from the PR doesn't magically make it fixed, it just takes my time, with no benefit for anyone.

Reply:
They way I see it, allowing tools/doctests to fail at all is a concession to the fact that synchronized updates are hard. But really a failing tool is not an acceptable state. If you are updating a tool and your tests are failing, that should be rejected just like when you are updating rustc and tests are failing.

Making sure tests pass is not a waste of time, it is a basic part of software maintenance.

Reply:
> As I said, I'm fine adding it back and seeing how it goes. I just don't want to spend much time dealing with failures going forward.

Thank you :) I will add it back. If it fails again, feel free to just drop rustc-guide and I can investigate.

> If you have spuriously failing tests, the precedent with RLS is to disable/ignore those tests on your side so that at least the other tests can still do their job.

Yes, this is the approach I have been taking with https://github.com/rust-lang/rustc-guide/pull/388. Also, Michael-F-Bryan [re-wrote the linkchecker](https://github.com/Michael-F-Bryan/mdbook-linkcheck/pull/11) to take better advantage of caching, and maybe it will have [more flexible handling of timeouts and server errors](https://github.com/Michael-F-Bryan/mdbook-linkcheck/issues/12) later too, so hopefully spurious failures will be less common in the future too.